js/donations-visualiser.js

```diff
@@ -855,49 +855,47 @@ function update(partyNodes, parties, selectedParties, resetControls) {
 
     d3.select("#party_select").selectAll(".checkbox").remove();
 
-    parties_grouped = []
-    parties_grouped_search = ["Coalition", "Australian Labor Party", "Australian Greens", "Palmer United Party"]
-    parties.map(function(item){
-      var index = parties_grouped_search.indexOf(party_map[item].name);
+    var major_party_names = ["Coalition", "Australian Labor Party", "Australian Greens", "Palmer United Party"];
+    // party indexes
+    var major_parties = [-1, -1, -1, -1];
```

Comment:
i think your're misunderstanding how arrays work in javascript. they're not like in most languages. in js they're sparse. so:
```
x = [];
x[23] = 'a';
x.forEach(console.log);
```
prints:
`a 34 [ , , , , , , , , , , , , , , , , , , , , , , , , , , , , , , , , , , 'a' ]`

i'm not sure why you're seeing that error, but it's not for that reason.

i checked the code by going to a year before the PUP was created and it worked fine.